### PR TITLE
add generic get method

### PR DESCRIPTION
**Pull request by niklongstone**
add a generic get  method to the Request class.
it is an alternative use of getQuery, getPost and getFile method.

- feature
- issues - none
- documentation - needed: nette/web-content#285
- BC break - no

### CODE REVIEW

File: src/Http/Request.php [hunk 1]
```diff
@@ -176,8 +177,34 @@ public function getCookies()
...
+    /**
+     * Provides a flexible generic get method.
+     * It follows an order of precedence: GET, POST, FILE.
+     *
+     * @param $key
+     * @param $default
+     *
+     * @return mixed
+     */
+    public function get($key, $default = NULL)
```
Comment (fprochazka) [inline src/Http/Request.php]: `$_REQUEST` all over again? :)

Comment (fprochazka) [inline src/Http/Request.php]: :-1:

Comment (JanTvrdik) [inline src/Http/Request.php]: It's safer that `$_REQUEST` because it does not allow cookies.

Comment (fprochazka) [inline src/Http/Request.php]: True :) But it still doesn't fee alright.

Comment (fprochazka) [inline src/Http/Request.php]: True :) But it still doesn't feel right.

Comment (Majkl578) [inline src/Http/Request.php]: `$_REQUEST` only includes cookies if `request_order` is misconfigured. It does not include them by default.

Comment (niklongstone) [inline src/Http/Request.php]: I think it worth it. 
This central method (is like a shortcut), the user won't be afraid about the HTTP verbs used in the Request, and could easily allow interface for other libraries.

**Comment by dg** [comment]
:-1:

**Comment by dg** [comment]
Thank you for your pull request, but I don't want to add this.

**Pull request closed without merging**